### DISCUSSION TITLE
minimal-bug-reports

### DISCUSSION THREAD

**Discussion by giscus** [mrocklin/mrocklin.github.io#13] [Giscus]
# minimal-bug-reports

Following up on a post on supporting users in open source this post lists some suggestions on how to ask a maintainer to help you with a problem. You don’t have to follow these suggestions. They ar...

https://matthewrocklin.com/minimal-bug-reports

**Comment by aidanheerdegen** [comment]
Hi,

Would it be ok to take a copy of this most excellent content (with a link back to your original)? 

Two reasons:

1. Things disappear/move
2. I'd like to adapt it slightly for a discourse setting

I did check for licensing, and noticed there was no explicit CC or similar, I assume on purpose, but no harm in asking eh?

Thanks

Aidan

**Reply by mrocklin** [reply]
Hey @aidanheerdegen , where were you thinking of making a copy?

**Reply by aidanheerdegen** [reply]
On a discourse forum we started to support climate modelling in Australia

https://forum.access-hive.org.au/

I wanted to make a nice MCVE topic for users when they're asking for help, and was very impressed with yours.

**Reply by mrocklin** [reply]
Cool.  Thanks for the content (and the implicit praise).  My recommendation is to link.  This is what other projects like Pandas do.  If you have a strong need to replicate that's fine, although I'd prefer that somehow you introduce a canonical-link metadata tag into the page if possible.  Otherwise search engines like Google start to observe duplicate content and ding both pages.

**Reply by aidanheerdegen** [reply]
No worries. I'll link.

**Reply by aidanheerdegen** [reply]
FYI: linked here https://forum.access-hive.org.au/t/how-to-create-a-minimal-complete-reproducible-example/843

**Comment by luifrancgom** [comment]
I understand that this may not be directly pertinent, but when introducing someone to a programming language and coding for the first time, how would you elucidate the process of crafting a Minimal, Complete, and Verifiable Example (MCVE)?

Grasping this concept can pose a significant challenge for novices in coding.

- Is it beneficial for them to attempt creating an MCVE, even if it means they might fail initially, and then use their specific issue as a case study to illustrate the construction of an MCVE?
- Might this hands-on, example-based learning approach aid in clarifying the subtleties mentioned in your guidance?

I pose these questions genuinely as I am about to lead an R programming course for individuals who have no prior coding experience, and your site offers exceptionally insightful advice on various topics.

**Comment by luifrancgom** [comment]
Regarding @aidanheerdegen's inquiry and considering the licensing terms specified in this repository concerning:

    The permission to reproduce, modify, distribute, perform, display, communicate, and translate a Work;
    The moral rights maintained by the original author(s) and/or performer(s);

What would be the recommended format for citing this post?

**Comment by dtomvan** [comment]
Just to let you know, the pandas community link (https://pandas.pydata.org/community.html) is dead. I'd replace it to their homepage or to https://github.com/pandas-dev/pandas?tab=readme-ov-file#getting-help Or just mention that they only really want you to use stackoverflow as of now.

In any case, definitely bookmarked so I can send it to the less bugreport-savvy :sweat_smile:

**Reply by mrocklin** [reply]
Thanks @dtomvan !  I've removed the link.

**Comment by kohlweypatrick-cmyk** [comment]
Thanks y'all, making fun is fired is the name of that babality
